### PR TITLE
Add shooted-seed documentation

### PR DESCRIPTION
**Pull request by dergeberl**
**How to categorize this PR?**
/area documentation
/kind enhancement
/priority normal

**What this PR does / why we need it**:
Documentation to create shooted-seeds with the annotation `shoot.gardener.cloud/use-as-seed`

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```improvement operator
NONE
```

### CODE REVIEW

File: docs/usage/shooted_seed.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Shooted Seeds
```
Comment (rfranzke) [inline docs/usage/shooted_seed.md:1]: Can you please add the document to the index (`docs/README.md`)?

File: docs/usage/shooted_seed.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+Create shooted seeds with the `shoot.gardener.cloud/use-as-seed` annotation
+
+## Preparation
+
+### 'garden' Project
```
Comment (rfranzke) [inline docs/usage/shooted_seed.md]:
```suggestion
### `garden` Project
```

File: docs/usage/shooted_seed.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+The annotation works only in the "garden" namespace. For that we have to create a project for the garden namespace by hand (the dashboard add "garden-" in the beginning).
```
Comment (rfranzke) [inline docs/usage/shooted_seed.md]:
```suggestion
The annotation works only for `Shoot`s created in the `garden` namespace. Consequently, we have to create a project for the `garden` namespace first using `kubectl` (we don't use the Gardener Dashboard as it would add a `garden` prefix for the namespace).
```

File: docs/usage/shooted_seed.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+example: <https://github.com/gardener/gardener/blob/master/example/05-project-dev.yaml>
```
Comment (rfranzke) [inline docs/usage/shooted_seed.md]:
```suggestion
Example: <https://github.com/gardener/gardener/blob/master/example/05-project-dev.yaml>
```

File: docs/usage/shooted_seed.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+kind: Project
+metadata:
+  name: garden
+spec:
+  owner:
+...
+  namespace: garden
+```
+
+### Edit 'garden' namespace
```
Comment (rfranzke) [inline docs/usage/shooted_seed.md]: Probably this should be the first step, or?

File: docs/usage/shooted_seed.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+labels:
+  gardener.cloud/role: project
+  project.gardener.cloud/name: garden
+```
+
+## Create shoot
+
+### Deploy shoot
+
+Unfortunately, the Gardener dashboard currently does not allow to change the CIDRs for the created shoot clusters via the form (its possible to edit the yaml directly in the UI), but your shoots won’t work if they have overlapping CIDR ranges with their corresponding seed cluster.
```
Comment (rfranzke) [inline docs/usage/shooted_seed.md]: Hm, the documentation about the Gardener Dashboard does not really belong into g/g. If you would like to have an e2e document then gardener/documentation is probably a better place.
For this PR/document, I would get rid of this paragraph and just describe what needs to be done to create a shooted seed.

File: docs/usage/shooted_seed.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    infrastructureConfig:
+      networks:
+        workers: 10.249.0.0/16
+...  
+  networking:
+    type: calico
+    pods: 10.96.0.0/11
+    nodes: 10.249.0.0/16
+    services: 10.64.0.0/13
+```
```
Comment (rfranzke) [inline docs/usage/shooted_seed.md]: These concrete values are unrelated to the document - you probably just have to mention that the networks shouldn't overlap, but this is usual business for seeds/shoots and concrete examples like these here are not needed. This document should focus on the annotation and which option it provides.

File: docs/usage/shooted_seed.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+Set the annotation for the shooted seed:
+
+```yaml
+  annotations:
+    shoot.gardener.cloud/use-as-seed: >-
+      true,shootDefaults.pods=100.96.0.0/11,shootDefaults.services=100.64.0.0/13,disable-capacity-reservation,with-secret-ref
+```
+
+This annotation contains a few configurations for the seed:
+
```
Comment (rfranzke) [inline docs/usage/shooted_seed.md]: Maybe a table provides a better overview than a list?

Comment (rfranzke) [reply docs/usage/shooted_seed.md]: There are more options, see https://github.com/gardener/gardener/blob/master/pkg/apis/core/v1beta1/helper/helper.go#L287-L369

File: docs/usage/shooted_seed.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+- true - enables the seed
```
Comment (rfranzke) [inline docs/usage/shooted_seed.md]:
```suggestion
- true - deploys the gardenlet into the shoot which will automatically register the cluster as seed
```

File: docs/usage/shooted_seed.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+- no-gardenlet - deploys no gardenlet on the shooted seed
```
Comment (rfranzke) [inline docs/usage/shooted_seed.md]:
```suggestion
- no-gardenlet - prevents the automatic deployment of the gardenlet into the shoot. Instead, the `Seed` object will be created with the assumption that another gardenlet will be responsible for managing it (according to its `seedSelector` configuration).
```

File: docs/usage/shooted_seed.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+- disable-capacity-reservation - set taint for disable-capacity-reservation (see <https://github.com/gardener/gardener/blob/master/example/50-seed.yaml>)
+- invisible - set taint for invisible (see <https://github.com/gardener/gardener/blob/master/example/50-seed.yaml>)
```
Comment (rfranzke) [inline docs/usage/shooted_seed.md]: The taints are deprecated in favor of the `.spec.settings`, so we probably should mention this instead of the taints (#2315)

File: docs/usage/shooted_seed.md [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+- with-secret-ref - adds a secretref in the seed (needed for dashboard's web terminals feature and to fetch the monitoring and logging credentials from the seed when logged in as operator)
```
Comment (rfranzke) [inline docs/usage/shooted_seed.md]:
```suggestion
- with-secret-ref - creates a secret with the kubeconfig of the cluster in the `garden` namespace in the garden cluster and specifies the `.spec.secretRef` in the `Seed` object accordingly.
```

File: docs/usage/shooted_seed.md [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+- shootDefaults.pods - network setting for shoots (see <https://github.com/gardener/gardener/blob/master/example/50-seed.yaml>)
```
Comment (rfranzke) [inline docs/usage/shooted_seed.md]:
```suggestion
- shootDefaults.pods - default pod network CIDR for shoots created on this seed
```

File: docs/usage/shooted_seed.md [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+- shootDefaults.services - network setting for shoots (see <https://github.com/gardener/gardener/blob/master/example/50-seed.yaml>)
```
Comment (rfranzke) [inline docs/usage/shooted_seed.md]:
```suggestion
- shootDefaults.services - default service network CIDR for shoots created on this seed
```

**Review by rfranzke — CHANGES_REQUESTED** [review]
/assign
Thanks for the PR, I have a few comments :)

File: docs/usage/shooted_seed.md (hunk 8 above)
Comment (dergeberl) [reply docs/usage/shooted_seed.md]: I add the options for the annotation. Unfortunately i don`t know the last 4 options in the table. Maybe you can help me out with this options?

File: docs/usage/shooted_seed.md (hunk 11 above)
Comment (dergeberl) [reply docs/usage/shooted_seed.md]: I edited the description for this options. Is there also a new setting for `seed.gardener.cloud/protected`?

File: docs/usage/shooted_seed.md [hunk 15]
```diff
@@ -0,0 +1,73 @@
...
+...
+  namespace: garden
+```
+
+## Create shoot
+
+### Deploy shoot
+
+Change the networks for the shooted seed that it is not overlapping with his shoots (spec.provider.infrastructureConfig.networks and spec.networking).
+Fullexample: [/example/90-shoot.yaml](../../example/90-shoot.yaml)
```
Comment (danielfoehrKn) [inline docs/usage/shooted_seed.md]:
```suggestion
Full example: [/example/90-shoot.yaml](../../example/90-shoot.yaml)
```

File: docs/usage/shooted_seed.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+
+Set the annotation for the shooted seed.
```
Comment (danielfoehrKn) [inline docs/usage/shooted_seed.md]:
```suggestion
Set the following  annotation on the shoot to mark it as a shooted seed.
```

File: docs/usage/shooted_seed.md [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+  owner:
+...
+  namespace: garden
+```
+
+## Create shoot
+
+### Deploy shoot
+
+Change the networks for the shooted seed that it is not overlapping with his shoots (spec.provider.infrastructureConfig.networks and spec.networking).
```
Comment (danielfoehrKn) [inline docs/usage/shooted_seed.md]: How can you "change"  the network of the shooted seed that is not created yet?
I guess that your intention is to create a `Seed` via shooted Seed,  that can have the same network ranges as Shoots running on the garden/root cluster. As such, this is an optional configuration.

If that is the intention of that section,  I would reformulate to make it a bit more clear.

File: docs/usage/shooted_seed.md [hunk 18]
```diff
@@ -0,0 +1,73 @@
...
+This annotation contains a few configurations for the seed:
+Option | Description
+--- | ---
+`true` | deploys the gardenlet into the shoot which will automatically register the cluster as seed
+`no-gardenlet` | prevents the automatic deployment of the gardenlet into the shoot. Instead, the `Seed` object will be created with the assumption that another gardenlet will be responsible for managing it (according to its `seedSelector` configuration).
+`disable-capacity-reservation` | set `spec.settings.excessCapacity.enabled` in Seed to false (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`invisible` | set `spec.settings.scheduling.visible` in Seed to false  (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`visible` | set `spec.settings.scheduling.visible` in Seed to true  (see [/example/50-seed.yaml](../../example/50-seed.yaml)) **default**
+`disable-dns` | set `spec.settings.shootDNS.enabled` in Seed to false  (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`protected` | only Sshoots in the `garden` namespace can use this seed
```
Comment (danielfoehrKn) [inline docs/usage/shooted_seed.md]:
```suggestion
`protected` | only Shoots in the `garden` namespace can use this seed
```

File: docs/usage/shooted_seed.md [hunk 19]
```diff
@@ -0,0 +1,73 @@
...
+`unprotected` | Shoots from all namespaces can use this seed **default**
+`with-secret-ref` | creates a secret with the kubeconfig of the cluster in the `garden` namespace in the garden cluster and specifies the `.spec.secretRef` in the `Seed` object accordingly.
+`shootDefaults.pods` | default pod network CIDR for shoots created on this seed
+`shootDefaults.services` | default service network CIDR for shoots created on this seed
+`minimumVolumeSize` | set `spec.volume.minimumSize` in Seed (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`blockCIDRs` | set `spec.network.blockCIDRs` seperated by `;` (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`backup.provider` | set `spec.backup.provider` in Seed (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`backup.region` | set `spec.backup.region` in Seed (see [/example/50-seed.yaml](../../example/50-seed.yaml))
```
Comment (danielfoehrKn) [inline docs/usage/shooted_seed.md]:
```suggestion
`backup.region` | set `spec.backup.region` in the Seed (see [/example/50-seed.yaml](../../example/50-seed.yaml))
```

File: docs/usage/shooted_seed.md [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+`visible` | set `spec.settings.scheduling.visible` in Seed to true  (see [/example/50-seed.yaml](../../example/50-seed.yaml)) **default**
+`disable-dns` | set `spec.settings.shootDNS.enabled` in Seed to false  (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`protected` | only Sshoots in the `garden` namespace can use this seed
+`unprotected` | Shoots from all namespaces can use this seed **default**
+`with-secret-ref` | creates a secret with the kubeconfig of the cluster in the `garden` namespace in the garden cluster and specifies the `.spec.secretRef` in the `Seed` object accordingly.
+`shootDefaults.pods` | default pod network CIDR for shoots created on this seed
+`shootDefaults.services` | default service network CIDR for shoots created on this seed
+`minimumVolumeSize` | set `spec.volume.minimumSize` in Seed (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`blockCIDRs` | set `spec.network.blockCIDRs` seperated by `;` (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`backup.provider` | set `spec.backup.provider` in Seed (see [/example/50-seed.yaml](../../example/50-seed.yaml))
```
Comment (danielfoehrKn) [inline docs/usage/shooted_seed.md]:
```suggestion
`backup.provider` | set `spec.backup.provider` in the Seed (see [/example/50-seed.yaml](../../example/50-seed.yaml))
```

File: docs/usage/shooted_seed.md [hunk 21]
```diff
@@ -0,0 +1,73 @@
...
+`backup.region` | set `spec.backup.region` in Seed (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`backup.secretRef.name` | set `spec.backup.secretRef.name` in Seed (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`backup.secretRef.namespace` | set `spec.backup.secretRef.namespace` in Seed (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`apiServer.autoscaler.minReplicas` |
```
Comment (danielfoehrKn) [inline docs/usage/shooted_seed.md]: If there is no description,  the `|` could be omitted?

File: docs/usage/shooted_seed.md (hunk 19 above)
Comment (danielfoehrKn) [reply docs/usage/shooted_seed.md]: Same nit in other sentences

File: docs/usage/shooted_seed.md [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+      true,shootDefaults.pods=100.96.0.0/11,shootDefaults.services=100.64.0.0/13,disable-capacity-reservation,with-secret-ref
+```
+
+This annotation contains a few configurations for the seed:
+Option | Description
+--- | ---
+`true` | deploys the gardenlet into the shoot which will automatically register the cluster as seed
+`no-gardenlet` | prevents the automatic deployment of the gardenlet into the shoot. Instead, the `Seed` object will be created with the assumption that another gardenlet will be responsible for managing it (according to its `seedSelector` configuration).
+`disable-capacity-reservation` | set `spec.settings.excessCapacity.enabled` in Seed to false (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`invisible` | set `spec.settings.scheduling.visible` in Seed to false  (see [/example/50-seed.yaml](../../example/50-seed.yaml))
```
Comment (danielfoehrKn) [inline docs/usage/shooted_seed.md]: Can we either always write `Seed` and `Shoot` in capital letters or always in small letters? There seem to be some irregularities.

**Review by danielfoehrKn — CHANGES_REQUESTED** [review]
Thanks for the documentation. 
My apologize for the late review, I have some additional comments.

File: docs/usage/shooted_seed.md (hunk 8 above)
Comment (rfranzke) [reply docs/usage/shooted_seed.md]: Which options exactly you need help with?

File: docs/usage/shooted_seed.md (hunk 11 above)
Comment (rfranzke) [reply docs/usage/shooted_seed.md]: Nope, for "protected" the taint will remain.

File: docs/usage/shooted_seed.md (hunk 17 above)
Comment (dergeberl) [reply docs/usage/shooted_seed.md]: Can you recheck this section please?

File: docs/usage/shooted_seed.md (hunk 21 above)
Comment (dergeberl) [reply docs/usage/shooted_seed.md]: I asked rfranzke to add a description for this options. I don`t know a description for that.

File: docs/usage/shooted_seed.md [hunk 23]
```diff
@@ -0,0 +1,75 @@
...
+`minimumVolumeSize` | set `spec.volume.minimumSize` in the Seed (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`blockCIDRs` | set `spec.network.blockCIDRs` seperated by `;` (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`backup.provider` | set `spec.backup.provider` in the Seed (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`backup.region` | set `spec.backup.region` in the Seed (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`backup.secretRef.name` | set `spec.backup.secretRef.name` in the Seed (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`backup.secretRef.namespace` | set `spec.backup.secretRef.namespace` in the Seed (see [/example/50-seed.yaml](../../example/50-seed.yaml))
+`apiServer.autoscaler.minReplicas` |
+`apiServer.autoscaler.maxReplicas` |
+`apiServer.replicas` |
+`use-serviceaccount-bootstrapping` |
```
Comment (dergeberl) [inline docs/usage/shooted_seed.md]: @rfranzke Can you help me with a description for this 4 options?

File: docs/usage/shooted_seed.md (hunk 8 above)
Comment (dergeberl) [reply docs/usage/shooted_seed.md]: I linked you in a comment on the 4 lines

File: docs/usage/shooted_seed.md (hunk 23 above)
Comment (rfranzke) [reply docs/usage/shooted_seed.md]: * `apiServer.replicas` controls how many kube-apiserver replicas the shooted seed gets by default
* `apiServer.autoscaler.minReplicas` controls the minimum number of kube-apiserver replicas for the shooted seed
* `apiServer.autoscaler.maxReplicas` controls the maximum number of kube-apiserver replicas for the shooted seed
* `use-serviceaccount-bootstrapping` states that the gardenlet shall register with the garden cluster using a temporary `ServiceAccount` instead of a `CertificateSigningRequest` (default)

**Review by rfranzke — DISMISSED** [review]
/lgtm
Thank you for your contribution!

**Review by danielfoehrKn — DISMISSED** [review]
/lgtm  Thanks looks good!  Can you squash the commits into one?

**Comment by rfranzke** [comment]
@dergeberl can you please squash the commits into a single one to make the PR ready for merge?

**Review by rfranzke — DISMISSED** [review]

**Comment by rfranzke** [comment]
Unfortunately, there are still conflicts that must be resolved via rebase.

**Review by rfranzke — APPROVED** [review]
